src/display/annotation_layer.js

```diff
@@ -38,341 +38,515 @@ var CustomStyle = displayDOMUtils.CustomStyle;
+      }
+    },
+
+    /**
+     * Show the content box.
+     *
+     * @private
+     * @param {boolean} pin
+     * @memberof TextAnnotationElement
+     */
+    _show: function TextAnnotationElement_show(pin) {
+      if (pin) {
+        this.pinned = true;
+      }
+      if (this.content.hasAttribute('hidden')) {
+        this.container.style.zIndex += 1;
+        this.content.removeAttribute('hidden');
+      }
+    },
+
+    /**
+     * Hide the content box.
+     *
+     * @private
+     * @param {boolean} unpin
+     * @memberof TextAnnotationElement
+     */
+    _hide: function TextAnnotationElement_hide(unpin) {
+      if (unpin) {
+        this.pinned = false;
+      }
+      if (!this.content.hasAttribute('hidden') && !this.pinned) {
+        this.container.style.zIndex -= 1;
+        this.content.setAttribute('hidden', true);
+      }
     }
+  });
+
+  return TextAnnotationElement;
+})();
+
+var WidgetAnnotationElement = (function WidgetAnnotationElementClosure() {
+  function WidgetAnnotationElement(parameters) {
+    AnnotationElement.call(this, parameters);
   }
 
-  function update(viewport, div, annotations) {
-    for (var i = 0, ii = annotations.length; i < ii; i++) {
-      var data = annotations[i];
-      var element = div.querySelector(
-        '[data-annotation-id="' + data.id + '"]');
-      if (element) {
-        CustomStyle.setProp('transform', element,
-          'matrix(' + viewport.transform.join(',') + ')');
+  Util.inherit(WidgetAnnotationElement, AnnotationElement, {
+    /**
+     * Render the widget annotation's HTML element in the empty container.
+     *
+     * @public
+     * @memberof WidgetAnnotationElement
+     */
+    render: function WidgetAnnotationElement_render() {
+      var content = document.createElement('div');
+      content.textContent = this.data.fieldValue;
+      var textAlignment = this.data.textAlignment;
+      content.style.textAlign = ['left', 'center', 'right'][textAlignment];
+      content.style.verticalAlign = 'middle';
+      content.style.display = 'table-cell';
+
+      var font = (this.data.fontRefName ?
+        this.page.commonObjs.getData(this.data.fontRefName) : null);
+      this._setTextStyle(content, font);
+
+      this.container.appendChild(content);
+      return this.container;
+    },
+
+    /**
+     * Apply text styles to the text in the element.
+     *
+     * @private
+     * @param {HTMLDivElement} element
+     * @param {Object} font
+     * @memberof WidgetAnnotationElement
+     */
+    _setTextStyle:
+        function WidgetAnnotationElement_setTextStyle(element, font) {
+      // TODO: This duplicates some of the logic in CanvasGraphics.setFont().
+      var style = element.style;
+      style.fontSize = this.data.fontSize + 'px';
+      style.direction = (this.data.fontDirection < 0 ? 'rtl': 'ltr');
+
+      if (!font) {
+        return;
       }
+
+      style.fontWeight = (font.black ?
+        (font.bold ? 'bolder' : 'bold') :
```

Comment:
I know that this is just existing code that is moved around, but anyway:
As we found out in PR #6091, using 'bolder' doesn't actually work as you'd probably expect, so if you want to, feel free to address this while you're touching the code.